Assets/Tests/InputSystem/CoreTests_Actions.cs

```diff
@@ -7072,4 +5788,194 @@ public void Actions_CanIterateOverActionsInMap()
 
         Assert.That(map.ToList(), Is.EquivalentTo(new[] { action1, action2, action3 }));
     }
+
+    [Test]
+    [Category("Actions")]
+    public void Actions_CanUseTouchWithActions()
+    {
+        InputSystem.settings.timesliceEvents = false;
+
+        var touchscreen = InputSystem.AddDevice<Touchscreen>();
+
+        var primaryTouchAction = new InputAction("PrimaryTouch" , "<Touchscreen>/primaryTouch/position");
+        var touch0Action = new InputAction("Touch0", "<Touchscreen>/touch0/position");
+        var touch1Action = new InputAction("Touch1", "<Touchscreen>/touch1/position");
+        var positionAction = new InputAction("Position", "<Touchscreen>/position");
+        var tapAction = new InputAction("Tap", "<Touchscreen>/tap");
+
+        Assert.That(primaryTouchAction.controls, Is.EquivalentTo(new[] { touchscreen.primaryTouch.position }));
+        Assert.That(touch0Action.controls, Is.EquivalentTo(new[] { touchscreen.touches[0].position }));
+        Assert.That(touch1Action.controls, Is.EquivalentTo(new[] { touchscreen.touches[1].position }));
+        Assert.That(positionAction.controls, Is.EquivalentTo(new[] { touchscreen.position }));
+        Assert.That(tapAction.controls, Is.EquivalentTo(new[] { touchscreen.tap }));
+
+        primaryTouchAction.Enable();
+        touch0Action.Enable();
+        touch1Action.Enable();
+        positionAction.Enable();
+        tapAction.Enable();
+
+        using (var trace = new InputActionTrace())
+        {
+            trace.SubscribeToAll();
+
+            BeginTouch(1, new Vector2(0.123f, 0.234f));
+
+            Assert.That(trace,
+                Started(primaryTouchAction, touchscreen.primaryTouch.position, new Vector2(0.123f, 0.234f))
+                    .AndThen(Performed(primaryTouchAction, touchscreen.primaryTouch.position, new Vector2(0.123f, 0.234f)))
+                    .AndThen(Started(positionAction, touchscreen.position, new Vector2(0.123f, 0.234f)))
+                    .AndThen(Performed(positionAction, touchscreen.position, new Vector2(0.123f, 0.234f)))
+                    .AndThen(Started(touch0Action, touchscreen.touches[0].position, new Vector2(0.123f, 0.234f)))
+                    .AndThen(Performed(touch0Action, touchscreen.touches[0].position, new Vector2(0.123f, 0.234f))));
+
+            trace.Clear();
+
+            BeginTouch(2, new Vector2(0.234f, 0.345f));
+
+            Assert.That(trace,
+                Started(touch1Action, touchscreen.touches[1].position, new Vector2(0.234f, 0.345f))
+                    .AndThen(Performed(touch1Action, touchscreen.touches[1].position, new Vector2(0.234f, 0.345f))));
+
+            trace.Clear();
+
+            EndTouch(1, new Vector2(0.345f, 0.456f));
+
+            Assert.That(trace,
```

Comment:
Q: is this what should happen? If I lift a finger, the secondary touch should drive the UI? Looking at native apps on my iPhone, that doesn't seem to be how it works.